Add Microsoft.Security views API spec

leonkosarev opened the pull request:
# Data Plane API Specification Update Pull Request

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.


<!-- 🚨🚨🚨 Replace this line with a summary and reason for these changes to your data plane API 🚨🚨🚨 -->

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

![spec_pr_review_workflow_diagram](https://github.com/Azure/azure-rest-api-specs/assets/4429827/5bb5e7ce-8aff-4dbb-a3f8-0d9b68fef5b1)

## API Info: The Basics

Most of the information about your service should be captured in the issue that serves as your [*API Spec engagement record*](https://aka.ms/azsdk/onboarding/restapischedule).

* Link to API Spec engagement record issue:

Is this review for (select one):

- [x] a private preview
- [ ] a public preview
- [ ] GA release

### Change Scope

<sup>This section will help us focus on the specific parts of your API that are new or have been modified. <br/>Please share a link to the design document for the new APIs, a link to the previous API Spec document (if applicable), and the root paths that have been updated. </sup>
* Design Document:
* Previous API Spec Doc:
* Updated paths:

### Viewing API changes

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

### Suppressing failures

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[Swagger-Suppression-Process](https://aka.ms/pr-suppressions) 
to get approval.

### Release planner

A [release plan](https://aka.ms/azsdkdocs/release-plans) should have been created. If not, please create one as it will help guide you through the REST API and SDK creation process. 

## ❔Got questions? Need additional info?? We are here to help!

<details>
  <summary> Contact us!</summary>

The [Azure API Review Board](https://aka.ms/azsdk/onboarding/restapischedule) is dedicated to helping you create amazing APIs. You can read about our mission and learn more about our process on our [wiki](https://aka.ms/azsdk/onboarding/restapischedule).
* 💬 [Teams Channel](https://teams.microsoft.com/l/channel/19%3a3ebb18fded0e47938f998e196a52952f%40thread.tacv2/General?groupId=1a10b50c-e870-4fe0-8483-bf5542a8d2d8&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47)
* 💌 [email](mailto://azureapirbcore@microsoft.com)

</details>

<details>
  <summary>Click here for links to tools, specs, guidelines & other good stuff</summary>

### Tooling

 * [Open API validation tools](https://aka.ms/swaggertools) were run on this PR. Go here to see [how to fix errors](https://aka.ms/ci-fix)
 * [Spectral Linting](https://github.com/Azure/azure-api-style-guide/blob/main/README.md)

### Guidelines & Specifications

 * [Azure REST API Guidelines](https://aka.ms/azapi/guidelines)
 * [OpenAPI Style Guidelines](https://aka.ms/azapi/style)
 * [Azure Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy)

### Helpful Links

 * [Schedule a data plane REST API spec review](https://aka.ms/azsdk/onboarding/restapischedule)

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom.
- If you don't have permissions to remove or add labels to the PR, request `write access` per [aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories](https://aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories) 
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

Commit:
Add Microsoft.Security API spec

Commit:
Add view example

rkmanda commented:
pl fix the Avocado errors for the review to continue
https://github.com/Azure/azure-rest-api-specs/pull/34531/checks?check_run_id=41876972456

rkmanda commented on specification/security/resource-manager/Microsoft.Security/preview/2025-05-01-preview/views.json:
```diff
@@ -0,0 +1,153 @@
+        }
+    },
+    "paths": {
+        "/providers/Microsoft.Security/views": {
```
>views [](http://example.com/codeflow?start=39&length=5)

rename to getView or fetchView or similar verb

leonkosarev replied:
Renamed to 'view'

rkmanda commented on specification/security/resource-manager/Microsoft.Security/preview/2025-05-01-preview/views.json:
```diff
@@ -0,0 +1,153 @@
+                        "in": "body",
+                        "required": true,
+                        "type": "object",
+                        "description": "The parameters for the specified scenario and view"
```
lets discuss why the scenario needs a generic schema

leonkosarev replied:
Scheduled a dedicated discussion on the topic

rkmanda commented on specification/security/resource-manager/Microsoft.Security/preview/2025-05-01-preview/views.json:
```diff
@@ -0,0 +1,153 @@
+                        "description": "Result of the query operation",
+                        "schema": {
+                            "$ref": "#/definitions/ScenarioResponse"
+                        }
```
lets discuss why the scenario needs a generic schema

Commit:
Update readme

Commit:
Update readme

Commit:
Add to default tag

Commit:
Add scopes parameter

Commit:
Move parameters

Commit:
Update

Commit:
Model validation

Commit:
Add properties

Commit:
Use schema

Commit:
Remove additionalProperties definition from views.json

Commit:
Refactor parameter definitions in views.json to inline schema descriptions

Commit:
Prettier

Commit:
Replace views.json with view.json and update API specifications

razvanbadea-msft commented:
Was the comments addressed? i see no updates for them

leonkosarev commented:
> Was the comments addressed? I see no updates for them

We addressed the comments - Resolving the mandatory checks - except for the net sdk generation which we saw fail in previous PR as well and we could not find an informative error for it.

As for the comment about the schema - we'd like to set a meeting with one of the reviewers to discuss.
For now, I scheduled office hours to get clarity on the above.

razvanbadea-msft commented on specification/security/resource-manager/Microsoft.Security/preview/2025-05-01-preview/view.json:
```diff
@@ -0,0 +1,176 @@
+          "type": "string",
+          "description": "The name of the scenario for which the data is requested"
+        },
+        "scenarioParameters": {
```
>scenarioParameters [](http://example.com/codeflow?start=9&length=18)

how could this be used? can you support any kind of properties for it? I see in the example that there is a startDateTime. Cant you add a schema for it for possible properties?
